src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -47,7 +48,19 @@ public WebClientDownloader(string userName, string password, ILogger logger)
 
             if (this.client == null)
             {
-                this.client = new HttpClient();
+                if (clientCertPath != null)
```

Comment:
Have you tried testing it in your scenario ? Does it work ?

Reply:
Yes, it is working as expected.